presto-main/src/main/java/com/facebook/presto/type/CharOperators.java

```diff
@@ -0,0 +1,205 @@
+import static com.facebook.presto.spi.function.OperatorType.EQUAL;
+import static com.facebook.presto.spi.function.OperatorType.GREATER_THAN;
+import static com.facebook.presto.spi.function.OperatorType.GREATER_THAN_OR_EQUAL;
+import static com.facebook.presto.spi.function.OperatorType.HASH_CODE;
+import static com.facebook.presto.spi.function.OperatorType.LESS_THAN;
+import static com.facebook.presto.spi.function.OperatorType.LESS_THAN_OR_EQUAL;
+import static com.facebook.presto.spi.function.OperatorType.NOT_EQUAL;
+import static com.facebook.presto.spi.type.StandardTypes.BOOLEAN;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+
+public final class CharOperators
+{
+    public static final SqlScalarFunction CHAR_EQUAL_NO_PAD = charLogicalOperator(EQUAL, "equalNoPad");
+    public static final SqlScalarFunction CHAR_NOT_EQUAL_NO_PAD = charLogicalOperator(NOT_EQUAL, "notEqualNoPad");
+    public static final SqlScalarFunction CHAR_LESS_THAN_NO_PAD = charLogicalOperator(LESS_THAN, "lessThanNoPad");
+    public static final SqlScalarFunction CHAR_LESS_THAN_OR_EQUAL_NO_PAD = charLogicalOperator(LESS_THAN_OR_EQUAL, "lessThanOrEqualNoPad");
+    public static final SqlScalarFunction CHAR_GREATER_THAN_NO_PAD = charLogicalOperator(GREATER_THAN, "greaterThanNoPad");
+    public static final SqlScalarFunction CHAR_GREATER_THAN_OR_EQUAL_NO_PAD = charLogicalOperator(GREATER_THAN_OR_EQUAL, "greaterThanOrEqualNoPad");
+    public static final SqlScalarFunction CHAR_BETWEEN_NO_PAD = charBetweenOperator("betweenNoPad");
+
+    public static final SqlScalarFunction CHAR_EQUAL_PAD_SPACES = charLogicalOperator(EQUAL, "equalPadSpaces");
+    public static final SqlScalarFunction CHAR_NOT_EQUAL_PAD_SPACES = charLogicalOperator(NOT_EQUAL, "notEqualPadSpaces");
+    public static final SqlScalarFunction CHAR_LESS_THAN_PAD_SPACES = charLogicalOperator(LESS_THAN, "lessThanPadSpaces");
+    public static final SqlScalarFunction CHAR_LESS_THAN_OR_EQUAL_PAD_SPACES = charLogicalOperator(LESS_THAN_OR_EQUAL, "lessThanOrEqualPadSpaces");
+    public static final SqlScalarFunction CHAR_GREATER_THAN_PAD_SPACES = charLogicalOperator(GREATER_THAN, "greaterThanPadSpaces");
+    public static final SqlScalarFunction CHAR_GREATER_THAN_OR_EQUAL_PAD_SPACES = charLogicalOperator(GREATER_THAN_OR_EQUAL, "greaterThanOrEqualPadSpaces");
+    public static final SqlScalarFunction CHAR_BETWEEN_PAD_SPACES = charBetweenOperator("betweenPadSpaces");
+
+    private CharOperators() {}
+
+    private static SqlScalarFunction charLogicalOperator(OperatorType operatorType, String... methods)
+    {
+        Signature signature = Signature.builder()
+                .kind(SCALAR)
+                .operatorType(operatorType)
+                .argumentTypes(
+                        parseTypeSignature("char(x)", ImmutableSet.of("x")),
+                        parseTypeSignature("char(y)", ImmutableSet.of("y")))
```

Comment:
If I define all comparison operators as `op(T,T)`, then it will not be possible to compare `CHARs` with `NO PADDING` (default) option, e.g: `CHAR(5) 'a    '` would be equal to `CHAR(1) 'a'`. This is inconsistent with how `VARCHARs` are compared.

 BTW: since https://github.com/prestodb/presto/pull/5717 is merged I think `op(char(x), char(y))` should work now.